internal/client.go

```diff
@@ -423,6 +425,10 @@ type (
 		// Default: 5s.
 		HealthCheckTimeout time.Duration
 
+		// HealthCheckRetryFor defines maximum amount of time that health check should be retried for.
+		// Defaults to 10s, once this timeout is reached error will be propagated to the client.
+		HealthCheckRetryFor time.Duration
```

Comment:
[Nit] I think we need better name here. OR, HealthCheckTimeout should represent the overall time including all attempts, and we have the new field specify the PerAttempt timeout (or have that hard-coded). I can see how this might cause some weirdness with breaking changes though...

Reply:
We can slightly turn it and add `HealthCheckAttemptTimeout`, which would allow us to use `HealthCheckTimeout` as an aggregate one, what do you think?